.github/workflows/tnf-image.yaml

```diff
@@ -106,6 +106,10 @@ jobs:
         with:
           ref: ${{ env.TNF_VERSION }}
 
+      - name: checkout the latest for the script directory in the main branch
+        run: |
+          git checkout origin/main -- Dockerfile
```

Comment:
IMHO, this is creating a complex mix of the stable and main branches when building the container image, and it can bite us in the future. We already have a similar pattern when checking out the create-version-files.sh and get-git-latest-release.sh scripts from main.

I think it might be easier to add a commit that fixes the Makefile to the 3.0.x branch, and then add a v3.0.1 tag including it. The GitHub action would then pick the new tag and use it to build the container, without the issues we are seeing today. 

I see that using Dockerfile from the main branch also pulls new Go and OC client versions. If that is what we want, we can also cherry-pick those changes to the 3.0.x branch.